zypp/media/MediaManager.cc

```diff
@@ -36,37 +37,91 @@ namespace zypp
     namespace // anonymous
     { ////////////////////////////////////////////////////////////////
 
+      struct ManagedMedia;
+      std::ostream & operator<<( std::ostream & str, const ManagedMedia & obj );
+
       // -------------------------------------------------------------
       struct ManagedMedia
       {
         ~ManagedMedia()
-        {}
+        {
+          try
+          {
+            close(); // !!! make sure handler gets properly deleted.
+          }
+          catch(...) {}
+        }
 
-        ManagedMedia()
-          : desired (false)
+        ManagedMedia( ManagedMedia &&m )
+          : desired ( m.desired )
+          , verifier( std::move(m.verifier) )
+          , _handler ( std::move(m._handler) )
         {}
 
-        ManagedMedia(const ManagedMedia &m)
-          : desired (m.desired)
-          , handler (m.handler)
-          , verifier(m.verifier)
-        {}
+        static ManagedMedia makeManagedMedia ( const Url &o_url, const zypp::filesystem::Pathname &preferred_attach_point, const MediaVerifierRef &v )
```

Comment:
```suggestion
        static ManagedMedia makeManagedMedia ( const Url &o_url, const Pathname &preferred_attach_point, const MediaVerifierRef &v )
```
+ all the cases I missed to see ;)

Reply:
Ok